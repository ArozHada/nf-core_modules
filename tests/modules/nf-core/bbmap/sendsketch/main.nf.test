nextflow_process {

    name "Test Process BBMAP_SENDSKETCH"
    script "modules/nf-core/bbmap/sendsketch/main.nf"
    process "BBMAP_SENDSKETCH"
    config "tests/modules/nf-core/bbmap/sendsketch/nextflow.config"
    tag "bbmap_sendsketch"

    test("Single End") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test', single_end:true], // meta map 
                             [ file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true) ]
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }

    }

}
