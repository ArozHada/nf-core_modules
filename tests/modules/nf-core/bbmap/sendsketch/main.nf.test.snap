{
    "Single End": {
        "content": [
            [
                "test.txt:md5,ba6a78659eb15d0d130d5daee8c87ead",
                "versions.yml:md5,694697087b868316aa450d1bc3775373"
            ]
        ],
        "timestamp": "2023-04-08T14:41:36+0000"
    }
}